go-sdk/internal/utxo/encryption_test.go

```diff
@@ -0,0 +1,206 @@
+	y, _ := new(big.Int).SetString("18264896395019517559018400396898398442219687903646821466907778802937824776999", 10)
+	key := []*big.Int{x, y}
+	nonce, _ := new(big.Int).SetString("220373351579243596212522709113509916796", 10)
+	c1, _ := new(big.Int).SetString("3623473636383738070031324804097049685564466189577273141109672613904615191520", 10)
+	c2, _ := new(big.Int).SetString("14192366070411288656840625597415252300006763456323771445497376523077328650161", 10)
+	c3, _ := new(big.Int).SetString("8948874854696341437962075211230225158124203775185169948359228967178039019393", 10)
+	c4, _ := new(big.Int).SetString("13654519867376896827561074824557193869787926453227340059166840999629617764240", 10)
+	cipherText := []*big.Int{c1, c2, c3, c4}
+	msg, err := PoseidonDecrypt(cipherText, key, nonce, 2)
+	assert.NoError(t, err)
+	assert.Equal(t, "1234567890", msg[0].Text(10))
+	assert.Equal(t, "2345678901", msg[1].Text(10))
+}
+
+func TestPoseidonEncryptDecryptWithLongMessages_3n(t *testing.T) {
+	nonce := NewEncryptionNonce()
+	x, _ := new(big.Int).SetString("14104886431895638088879904796248988944763544789684292755064599086710742631244", 10)
+	y, _ := new(big.Int).SetString("12567888666920372522142016384715158971908391387943244674769979344082830343251", 10)
+	sharedKey := []*big.Int{x, y}
+
+	// Encrypt a message of length 3n but bigger than 3
+	msg := []*big.Int{big.NewInt(1234567890), big.NewInt(2345678901), big.NewInt(3456789012), big.NewInt(4567890123), big.NewInt(5678901234), big.NewInt(6789012345)}
+	cipherText, err := PoseidonEncrypt(msg, sharedKey, nonce)
+	assert.NoError(t, err)
+	assert.Equal(t, 7, len(cipherText))
+
+	// Decrypt the message
+	decryptedMsg, err := PoseidonDecrypt(cipherText, sharedKey, nonce, 6)
+	assert.NoError(t, err)
+	assert.Equal(t, 6, len(decryptedMsg))
+	assert.Equal(t, "1234567890", decryptedMsg[0].Text(10))
+	assert.Equal(t, "2345678901", decryptedMsg[1].Text(10))
+	assert.Equal(t, "3456789012", decryptedMsg[2].Text(10))
+	assert.Equal(t, "4567890123", decryptedMsg[3].Text(10))
+	assert.Equal(t, "5678901234", decryptedMsg[4].Text(10))
+	assert.Equal(t, "6789012345", decryptedMsg[5].Text(10))
+}
+
+func TestPoseidonEncryptDecryptWithLongMessages_3nPlus1(t *testing.T) {
+	nonce := NewEncryptionNonce()
+	x, _ := new(big.Int).SetString("14104886431895638088879904796248988944763544789684292755064599086710742631244", 10)
+	y, _ := new(big.Int).SetString("12567888666920372522142016384715158971908391387943244674769979344082830343251", 10)
+	sharedKey := []*big.Int{x, y}
+
+	// Encrypt a message of length 3n but bigger than 3
+	msg := []*big.Int{big.NewInt(1234567890), big.NewInt(2345678901), big.NewInt(3456789012), big.NewInt(4567890123), big.NewInt(5678901234), big.NewInt(6789012345), big.NewInt(7890123456)}
+	cipherText, err := PoseidonEncrypt(msg, sharedKey, nonce)
+	assert.NoError(t, err)
+	assert.Equal(t, 10, len(cipherText))
+
+	// Decrypt the message
+	decryptedMsg, err := PoseidonDecrypt(cipherText, sharedKey, nonce, 7)
```

Comment:
wait wouldn't you want to test with a different shared key so `public B, private A` encrypt and `public A, private B` decrypt?

Reply:
using ecdh to derive the shared secret is performed in the integration-test under Test 2. here we are focused on the encryption and decryption, which is a symmetric scheme so the same key is used for both